### ISSUE TITLE
how about Backpropagation

### ISSUE THREAD

**Issue by jjhesk** [T-Specht/thoughtful#2]
is that ready?

**Comment by T-Specht** [comment]
Yep, it is ready and implemented, even though without much customization at the moment. You can only choose the learning rate yourself at the moment. Momentum is not yet working that well. 
But the Neural Network class will automatically use backpropagation behind the scenes when training so you do not have to worry about it 👍🏼
